pkg/util/parquet/write_functions.go

```diff
@@ -264,13 +264,11 @@ type writeFn func(d tree.Datum, w file.ColumnChunkWriter, a *batchAlloc, fmtCtx
 	defLevels, repLevels []int16) error
 
 // formatDatum writes the datum into the parquet.ByteArray batch alloc using the
-// tree.NodeFormatter interface. It is important that the fmtCtx remains open
-// until after the bytes have been read from the batchAlloc, otherwise the byte
-// slice may point to invalid data.
+// tree.NodeFormatter interface.
 func formatDatum(d tree.Datum, a *batchAlloc, fmtCtx *tree.FmtCtx) {
 	fmtCtx.Reset()
 	d.Format(fmtCtx)
-	a.byteArrayBatch[0] = fmtCtx.Bytes()
+	a.byteArrayBatch[0] = fmtCtx.CopyBytes()
```

Comment:
yes, i think you should do it this way (call `tree.NewFmtCtx` rather than passing one in. it will be easier to use safely, and `NewFmtCtx` already makes use of a `sync.Pool` that contains `FmtCtx` objects, so there aren't as many allocations.

Reply:
This all makes sense to me. `fmtCtx.CloseAndGetString()` is the only allocation we incur here.